index.js

```diff
@@ -24,18 +25,83 @@ db.once('open', function() {
   console.log('Spojeni smo na MongoDB bazu');
 });
 
-const { Schema } = mongoose;
-
-const userSchema = new Schema({
+const shema = new mongoose.Schema({
   username: { type: String, required: true },
   email: { type: String, required: true },
   password: { type: String, required: true },
+})
+
+const primjerKorisnika = mongoose.model('User', shema)
+
+app.use(bodyParser.json());
+
+app.post('mongodb://127.0.0.1:27017/myFitness', async (req, res) => {
+  try {
+    
+    const { username, mail, lozinka } = req.body;
+
+    
+    const noviKorisnik = new primjerKorisnika({ username, mail, lozinka });
+
+    
+    await noviKorisnik.save();
+
+    //šaljen odg na frontend
+    res.status(200).json({ success: true, message: 'Registracija je uspješna!' });
+
+    //kreiran token da bi korisnik ostao logiran
+    const token = jwt.sign({ username }, 'tajna', { expiresIn: '1h' });
+
+    //šaljen token natrag na frontend
+    res.status(200).json({ success: true, message: 'Prijava uspješna!', token });
+  } catch (error) {
+    console.error('Greška prilikom registracije:', error);
+    res.status(500).json({ success: false, message: 'Došlo je do pogreške prilikom registracije.' });
+  }
 });
 
-const User = mongoose.model("User", userSchema);
-const newUser = new User({
-  username: "mate",
-  email: 'mate@pmfst.hr',
-  password: "ohvaho490vs",
+//do tu registracija
+
+//odavde prijava
+
+const bcrypt = require('bcrypt');
+
+app.post('mongodb://127.0.0.1:27017/myFitness', async (req, res) => {
+  try {
+    const { username, lozinka } = req.body;
+
+    //traženje korisnika po imenu
+    const pronadjeniKorisnik = await primjerKorisnika.findOne({ username });
```

Comment:
jel mozemo cili login obavit u jednom pozivu prima bazi, vidit jel ima odgovarajuci username i password u bazi i ako nema vratit gresku da su podaci netocni
Pozivi pribma bazi su naskuplji, to jest trose najvise vremena i usporavaju citavu app

Reply:
Ne moramo nuzno ovako napravit, mozes probat i vidit, pvu logiku mozemo minja ti kasnije ne narusava funkcionalnost vise je optimizacija

Reply:
pogledat cu